### PR TITLE
Data usage

### PR DESCRIPTION
**Pull request by bwmarkle**

### CODE REVIEW

File: src/assets/js/usage.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+		/**
+		 * Get the user's client id.
+		 *
+		 * @since SINCEVERSION
+		 */
+		getClientId: function() {
+			var clientId = '';
+
+			ga.getAll().forEach(tracker => {
```
Comment (cssjoe) [inline src/assets/js/usage.js]: `usage.js?ver=20191213:80 Uncaught TypeError: ga.getAll is not a function
    at Object.getClientId (usage.js?ver=20191213:80)
    at Object.prependClientId (usage.js?ver=20191213:42)
    at HTMLDocument.<anonymous> (usage.js?ver=20191213:166)`

Comment (bwmarkle) [reply src/assets/js/usage.js]: This has been fixed in another commit - https://github.com/BoldGrid/library/commit/1fd4601d27cfec75b67fe2a98e1e6d69449d6b4d

File: src/Library/Usage/Notice.php [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	}
+
+	/**
+	 * Handle the ajax call to accept / decline the notice.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public static function wp_ajax_bglib_usage_signup() {
+		if( ! check_ajax_referer( 'bglib_usage_signup', 'nonce', false ) ) {
+			wp_send_json_error( __( 'Invalid nonce.', 'boldgrid-library' ) );
```
[the remark this answers is not in the document]
Comment (bwmarkle) [reply src/Library/Usage/Notice.php]: Updated - https://github.com/BoldGrid/library/commit/e7e2a76516ba748392d8e8d7bcb09db88f309bda

File: src/Library/Usage.php [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	private static $filtersAdded = false;
+
+	/**
+	 * GA ID.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var string
+	 */
+	private $gaId = 'UA-1501988-42';
```
[the remark this answers is not in the document]
Comment (bwmarkle) [reply src/Library/Usage.php]: Fixed - https://github.com/BoldGrid/library/commit/2f287093bb58d26c6d3ecafdfe256b1314e8517e

File: src/Library/Usage/Helper.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+class Helper {
+	/**
+	 * Whether or not page in the admin.php?page= begins with a prefix.
+	 *
+	 * @return bool True if the page begins with one of the prefixes.
+	 */
+	public static function hasPagePrefix() {
+		$prefixes      = apply_filters( 'Boldgrid\Library\Usage\getPrefixes', [] );
+		$hasPagePrefix = false;
+		$page          = ! empty( $_GET['page'] ) ? $_GET['page'] : null;
```
[the remark this answers is not in the document]
Comment (bwmarkle) [reply src/Library/Usage/Helper.php]: Fixed in https://github.com/BoldGrid/library/commit/714705940cfd7b497021c18f737a2a52b88109ec and https://github.com/BoldGrid/boldgrid-backup/commit/3f168879013f9edee233dfb2fdce25c4608e6444

**Pull request merged**